lib/Processor.php

```diff
@@ -1193,26 +1171,23 @@ private function sendHandshake($inSSL = false) {
         $payload .= str_repeat("\0", 23); // reserved
 
         if (!$inSSL && ($this->capabilities & self::CLIENT_SSL)) {
-            $this->_sendPacket($payload);
-            Loop::onWritable($this->socket, function ($watcherId, $socket) {
-                /* wait until main write watcher has written everything... */
-                if ($this->outBuflen > 0 || !empty($this->out)) {
+            \Amp\call(function () use ($payload) {
+                yield $this->write($payload);
+
+                $context = $this->config->ssl ?: new ClientTlsContext;
+                $context = $context->withPeerName($this->config->host);
+
+                return yield $this->socket->enableCrypto($context);
+            })->onResolve(function ($error) {
```

Comment:
Why is that inside `onResolve` and not directly inside the coroutine?

Reply:
Oversight when refactoring. This onResolve function existed in the prior version at [line 1205-1210](https://github.com/amphp/mysql/pull/49/files/604fd53853a69dc133f48ee41a3d0098cbf38788#diff-b7c6c28a97dddfea08068fa97b76dafdL1205). I'll roll it into the coroutine.